test/gtest/ucp/test_ucp_atomic.cc

```diff
@@ -28,6 +30,11 @@ class test_ucp_atomic : public test_ucp_memheap {
         add_variant_with_value(variants, features, variant | UCP_ATOMIC_MODE_CPU, "cpu" + name);
         add_variant_with_value(variants, features, variant | UCP_ATOMIC_MODE_DEVICE, "device" + name);
         add_variant_with_value(variants, features, variant | UCP_ATOMIC_MODE_GUESS, "guess" + name);
+
+        // Do not test device atomics with ODP since SW atomic emulation
+        // disabled with device atomics and there can be configurations
+        // without atomic lanes with ODP support which would lead to failure
```

Comment:
```suggestion
        /* Do not force using device atomics when testing ODP, since some
         * configurations do not support it. Allow a fallback to SW emulation
         * in this case.
         */
```